multisig/src/lib.rs

```diff
@@ -107,22 +134,16 @@ impl MultiSigContract {
     }
 
     /// Remove given request and associated confirmations.
-    pub fn delete_request(&mut self, request_id: RequestId) {
-        assert_eq!(
-            env::current_account_id(),
-            env::predecessor_account_id(),
-            "Predecessor account must much current account"
-        );
-        assert!(
-            self.requests.get(&request_id).is_some(),
-            "No such request: either wrong number or already confirmed"
-        );
-        assert!(
-            self.confirmations.get(&request_id).is_some(),
-            "Internal error: confirmations mismatch requests"
-        );
-        self.requests.remove(&request_id);
-        self.confirmations.remove(&request_id);
+    pub fn delete_request(&mut self, request_id: RequestId) -> bool {
+        self.valid_request(request_id);
```

Comment:
may be then `let added = self.get_valid_request_or_fail(request_id).added` ?

because actually there is no reason to do `assert!(
            self.requests.get(&request_id).is_some(),` given `self.requests.get(&request_id).expect(..)` will do the same.